rest/src/main/scala/io/udash/rest/openapi/RestStructure.scala

```diff
@@ -67,23 +67,43 @@ object RestStructure extends AdtMetadataCompanion[RestStructure] {
     @checked @infer restSchema: RestSchema[T],
     @composite info: GenCaseInfo[T]
   ) extends Case[T] {
+
     def caseSchema(caseFieldName: Opt[String]): RestSchema[T] =
       caseFieldName.fold(restSchema) { cfn =>
         val caseFieldSchema = RefOr(Schema.enumOf(List(info.rawName)))
-        val taggedName =
-          if (restSchema.name.contains(info.rawName)) s"tagged${info.rawName}"
-          else info.rawName
-        restSchema.map({
-          case RefOr.Value(caseSchema) => caseSchema.copy(
-            properties = caseSchema.properties + (cfn -> caseFieldSchema),
-            required = cfn :: caseSchema.required
-          )
-          case ref => Schema(allOf = List(RefOr(Schema(
-            `type` = DataType.Object,
-            properties = IListMap(cfn -> caseFieldSchema),
-            required = List(cfn)
-          )), ref))
-        }, taggedName)
+
+        def schemaWithDiscriminatorField: RestSchema[T] = {
+          val taggedName = if (restSchema.name.contains(info.rawName)) s"tagged${info.rawName}" else info.rawName
+          restSchema.map({
+            case RefOr.Value(caseSchema) => caseSchema.copy(
+              properties = caseSchema.properties + (cfn -> caseFieldSchema),
+              required = cfn :: caseSchema.required
+            )
+            case ref => Schema(allOf = List(RefOr(Schema(
+              `type` = DataType.Object,
+              properties = IListMap(cfn -> caseFieldSchema),
+              required = List(cfn)
+            )), ref))
+          }, taggedName)
+        }
+
+        restSchema.createSchema(ShallowInliningResolver) match {
```

Comment:
`ShallowInliningResolver` could be private somewhere close to this code (e.g. private object in `RestStructure`)